api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java

```diff
@@ -198,4 +239,167 @@ public String getComplexObsDir() {
 	public String getTestComplexObsFilePath() {
 		return getComplexObsDir() + "/" + getTestFileNameWithExt();
 	}
+	
+	/**
+	 * Boilerplate method to save a collection of complex obs.
+	 *
+	 * @param count The number of the complex obs to be saved.
+	 */
+	public List<Obs> saveComplexObs(int count) throws IOException {
+		
+		init();
+		
+		List<Obs> obsList = new ArrayList<>();
+		byte[] randomData = new byte[20];
+		
+		Patient patient = Context.getPatientService().getPatient(2);
+		Visit visit = Context.getVisitService().getVisit(1);
+		
+		EncounterService encounterService = Context.getEncounterService();
+		EncounterType encounterType = encounterService.getEncounterType(1);
+		Provider provider = Context.getProviderService().getProvider(1);
+		context.getAdministrationService().saveGlobalProperty(
+		    new GlobalProperty(AttachmentsConstants.GP_ENCOUNTER_TYPE_UUID, encounterType.getUuid()));
+		Encounter encounter = context.getAttachmentEncounter(patient, visit, provider);
+		
+		for (int i = 0; i < count; i++) {
+			String fileCaption = RandomStringUtils.randomAlphabetic(12);
+			new Random().nextBytes(randomData);
+			MockMultipartFile multipartRandomFile = new MockMultipartFile(String.valueOf(i), String.valueOf(i),
+			        "application/octet-stream", randomData);
+			obsList.add(obsSaver.saveOtherAttachment(visit, patient, encounter, fileCaption, multipartRandomFile,
+			    ValueComplex.INSTRUCTIONS_DEFAULT));
+		}
+		return obsList;
+	}
+	
+	/**
+	 * Boilerplate method to save a collection of complex obs.
+	 *
+	 * @param count The number of the complex obs to be saved.
+	 */
+	public List<Obs> saveComplexObs(Encounter encounter, int count, int otherCount, boolean allComplexRelatedToAttach)
+	        throws IOException {
+		init();
+		
+		// Creating the list of Complex Obs relevnt to the attachment module (
+		// Attachments ) using ComplexObsSaver class.
+		List<Obs> obsList = new ArrayList<>();
+		List<Obs> otherObsList = new ArrayList<>();
+		byte[] randomData = new byte[20];
+		Patient patient = encounter.getPatient();
+		Visit visit = encounter.getVisit();
+		for (int i = 0; i < count; i++) {
+			String fileCaption = RandomStringUtils.randomAlphabetic(12);
+			new Random().nextBytes(randomData);
+			MockMultipartFile multipartRandomFile = new MockMultipartFile(String.valueOf(i), String.valueOf(i),
+			        "application/octet-stream", randomData);
+			obsList.add(obsSaver.saveOtherAttachment(visit, patient, encounter, fileCaption, multipartRandomFile,
+			    ValueComplex.INSTRUCTIONS_DEFAULT));
+		}
+		for (int i = 0; i < otherCount; i++) {
```

Comment:
This second loop up to `otherCount` is specific to adding non-attachments complex obs. "Oher" refers to complex obs that are coded with a concept that is _other than for attachments_.

Does that make sense?

Reply:
OK , I understood it wrongly. Sorry. I will quickly changed it and update the PR.